src/test/shared/utilities/assertUtils.ts

```diff
@@ -23,3 +23,27 @@ export async function assertRejects(action: () => Promise<any>) {
         })
     }
 }
+
+export async function assertThrowsError(
```

Comment:
```typescript
...
} catch (err) {
    error = err as Error
} finally {
...
```

Just checking: Are you expecting this to assert that `err` is an `Error`? In C#, casts have type information available at runtime, so they can detect invalid casts. In TypeScript, `as Error` just instructs the compiler to assume that the object has `Error`'s shape. It won't throw if the object does not have `Error`'s shape.

Suggest to change to either `let error: any ... error = err` (since anything can be thrown in JS), or `assert.ok(err instanceof Error)` (if you want to ensure that an `Error` was thrown).

Reply:
The intention is to use this in tests where we are checking for explicit `Error` objects. Adding the assert.ok check.